wordpress-mobile/WordPress-Android
# Add seen/unseen support to all posts.

Depends on #13836

This PR expands the seen/unseen toggle support to all the posts made before 2020-07-13

What changed:
- Updated the main endpoint for toggling seen/unseen status. Now we are using post and blog ID's to toggle the status.
- We use the previous endpoint to toggle seen/unseen status of non-wpcom posts.
- Instead of using `feed_item_id` to decide either to show the menu button or not we are using absence of `is_seen` flag to determine this. Note: this field is still missing from some endpoints (like tags and search) so until it's there I will be using a date comparison to determine if the feature if available for a specific post.

Note:
- Seen/Unseen button will not be visible right after DB upgrade, since we need to update posts and populate `is_seen` and `is_seen_supported` flag into DB.
- I found a couple of posts where seen/unseen toggle is not working. It also doesn't work on calypso for same posts, so there is not much we can do on client side until this is fixed. 

To test:
- Enable `SEEN_UNSEEN_WITH_COUNTER` feature and refresh posts.
- Confirm that the Seen/Unseen functionality works with all the posts in Reader - Followed, Discover, Blog Preview, P2, Likes, Tags.
- Confirm that the Seen/Unseen functionality works with non-wpcom posts (eg. from cnn.com).
- Confirm that the posts created before 2020-07-13 do not have a Seen/Unseen button in the post menu. It's easy to get to old posts in discover.wordpress.com or with some less popular tags/topics. 

PR submission checklist:
